### PR TITLE
Analytics posts

### PR DESCRIPTION
**Pull request by deviantintegral**
Initial support for loading posts: https://www.parse.ly/help/api/analytics/

Key features:

* Basic testing infrastructure, with support for debugging requests during functional tests.
* Uses Symfony's serializer to put all of the data into a PostList and Post object.
* Adds CircleCI for tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# PHP CircleCI 2.0 configuration file
```
Comment (deviantintegral) [inline .circleci/config.yml]: This is pretty much directly copied from https://github.com/Lullabot/mpx-php/blob/master/.circleci/config.yml

File: src/Post.php [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+/**
+ * Represents a Post.
+ *
+ * @todo Is this data model shared for all posts? The API is not REST so it's
+ * hard to tell until we need to load posts from another resource.
+ *
+ * @see https://www.parse.ly/help/api/analytics/
+ */
+class Post
```
Comment (deviantintegral) [inline src/Post.php]: All the methods from here on are the various properties documented by Parse.ly. There's nothing more than generated getters and setters.

File: src/PostList.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Lullabot\Parsely;
+
+/**
+ * Class PostList.
+ *
+ * @todo Do all lists of any object have the same format? If so, this could
+ *   become just 'List'.
+ */
+class PostList implements \ArrayAccess, \Countable
```
Comment (deviantintegral) [inline src/PostList.php]: This exists so we can extend it later to support transparent paging. That may mean we have to scale this back to `\Iterator`, but the idea is if you have a post list of 10000 results with a limit set set to 50, every time you `foreach` over a page boundary we can transparently fetch it.

File: tests/src/Functional/Analytics/PostsTest.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+use Lullabot\Parsely\PostList;
+use Lullabot\Parsely\Tests\Functional\FunctionalTestBase;
+
+class PostsTest extends FunctionalTestBase
+{
+
+    /**
+     * A test that loads 10 posts.
+     */
+    public function testExecute()
```
Comment (deviantintegral) [inline tests/src/Functional/Analytics/PostsTest.php]: This test is helpful for debugging the API. Follow the steps in `phpunit.xml.dist` and run this test in PHPUnit or at the CLI with `vendor/bin/phpunit --testsuite functional` and you'll get something like this:

poststest php parse ly-php 2018-09-14 14-45-43

Comment (deviantintegral) [reply tests/src/Functional/Analytics/PostsTest.php]: Hm show this again?

**Comment by juampynr** [comment]
Looks good. Merging.

**Pull request merged**